nevergrad/optimization/optimizerlib.py

```diff
@@ -1451,9 +1452,18 @@ def _learn_on_k_best(archive: utils.Archive[utils.MultiValue], k: int) -> tp.Arr
     model = LinearRegression()
     model.fit(X2, y)
 
+    # Check model quality.
+    model_outputs = model.predict(X2)
+    indices = np.argsort(y)
+    ordered_model_outputs = [model_outputs[i] for i in indices]
+    if not all(x < y for x, y in zip(ordered_model_outputs, ordered_model_outputs[1:])):
```

Comment:
This is much simpler (just need to verify the it's >0 and not <0 though, since I'm not sure I understand the aim here)
```suggestion
    if not np.all(np.diff(ordered_model_outputs) > 0):
```
Can you clarify the aim of this check?